phpseclib/Net/SSH2.php

```diff
@@ -2533,6 +2539,42 @@ public function exec(string $command, callable $callback = null)
         }
     }
 
+    /**
+     * Opens a channel
+     */
+    protected function openChannel(string $channel, bool $skip_extended = false): bool
+    {
+        $this->channelCount++;
```

Comment:
Also, if meant to track number of open channels, I do not see it decremented ever.

Reply:
You wouldn't necessarily have to `unset($this->channel_status[$channel])`. An alternative would be to inspect the actual status values and to make a determination from there, rather than simply tallying up the number of channel status entries.

Reply:
I guess you could do `count(array_filter($this->channel_status, function($val) { return $val != NET_SSH2_MSG_CHANNEL_CLOSE; }))` but, as a matter of personal preference, I prefer `$this->channelCount`.

If this was a PR being done by someone else and they did `array_filter` instead of what I'm doing I prob wouldn't comment on it.

Reply:
Right, and it can work either way. My preference for `channel_status` is just in that it involves less state, and mgmt thereof, than the additional `channel_count`.